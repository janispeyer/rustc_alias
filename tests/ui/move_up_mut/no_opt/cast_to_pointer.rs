// compile-flags: -Zmir-emit-retag

fn move_up_mut_no_opt3(x: &mut i32, y: &mut i32) -> i32 {
    *x = 42;
    let z = x as *mut i32;
    *y = 7;
    return *x;
}

fn main() {
    let mut x = 1;
    let mut y = 2;
    move_up_mut_no_opt3(&mut x, &mut y);
}
