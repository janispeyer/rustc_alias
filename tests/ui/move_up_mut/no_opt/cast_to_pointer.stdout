# Analysing main
# Analysing cast_to_pointer
# MaybeTopOfBorrowStack Analysis
bb0
[ 0] [_1, _2] <- Retag([fn entry] _1)
[ 1] [_1, _2] <- Retag([fn entry] _2)
[ 2] [_1, _2] <- (*_1) = const 42_i32
[ 3] [_1, _2] <- StorageLive(_4)
[ 4] [_2] <- _4 = &raw mut (*_1)
[ 5] [_2] <- Retag([raw] _4)
[ 6] [_2] <- (*_2) = const 7_i32
[ 7] [_2] <- _0 = (*_1)
[ 8] [_2] <- StorageDead(_4)
[ 9] [] <- return

# FindImmutabilitySpans Analysis
bb0
[ 0] {} <- Retag([fn entry] _1)
[ 1] {} <- Retag([fn entry] _2)
[ 2] {_1: Span(bb0[2])} <- (*_1) = const 42_i32
[ 3] {_1: Span(bb0[2])} <- StorageLive(_4)
[ 4] {_1: Top} <- _4 = &raw mut (*_1)
[ 5] {_1: Top} <- Retag([raw] _4)
[ 6] {_1: Top, _2: Span(bb0[6])} <- (*_2) = const 7_i32
[ 7] {_1: Top, _2: Span(bb0[6])} <- _0 = (*_1)
[ 8] {_1: Top, _2: Span(bb0[6])} <- StorageDead(_4)
[ 9] {_1: Top, _2: Span(bb0[6])} <- return


# CFG Before
bb0
Retag([fn entry] _1)
Retag([fn entry] _2)
(*_1) = const 42_i32
StorageLive(_4)
_4 = &raw mut (*_1)
Retag([raw] _4)
(*_2) = const 7_i32
_0 = (*_1)
StorageDead(_4)
return

# CFG After
bb0
Retag([fn entry] _1)
Retag([fn entry] _2)
(*_1) = const 42_i32
StorageLive(_4)
_4 = &raw mut (*_1)
Retag([raw] _4)
(*_2) = const 7_i32
_0 = (*_1)
StorageDead(_4)
return

